tests/Services/SearchServiceTest.php

```diff
@@ -1283,6 +1283,45 @@ public function testInitialNavigationPageGetsResultCountFromPlenty()
         $searchService->doNavigation($requestMock, $externalSearchServiceMock);
     }
 
+    public function testRetryMechanismAndEnsureItGetsLogged()
+    {
+        $requestMock = $this->getMockBuilder(HttpRequest::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $externalSearchServiceMock = $this->getMockBuilder(ExternalSearch::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $externalSearchServiceMock->expects($this->never())->method('setResults');
+
+        $this->requestBuilder->expects($this->once())->method('build')
+            ->willReturn(new Request());
+
+        $plentyErrorResponse = [
+            'error' => true,
+            'error_msg' => 'no services found',
+            'error_file' => '/var/www/SdkRestApi.php',
+        ];
+        $nonStringErrorResponse = false;
+        $validResponse = $this->getMockResponse('someResultsWithFilters.xml');
+
+        $this->client->expects($this->exactly(3))
+            ->method('call')
+            ->willReturnOnConsecutiveCalls($plentyErrorResponse, $nonStringErrorResponse, $validResponse);
+
+        $this->logger->expects($this->at(0))->method('error')->with(
```

Comment:
`at()` [is deprecated](https://github.com/sebastianbergmann/phpunit/issues/4297). Use `exactly()` and `withConsecutive()` instead. Similar how you already did it in line `1308`.